### PR TITLE
Bug fix + improvement

### PR DESCRIPTION
**Pull request by ali-bahjati**
This PR mostly fixes some logic bugs or improve them. Also updates the calendar for the new year.

Please see my comments for further explanation.

### CODE REVIEW

File: pyth_observer/__init__.py [hunk 1]
```diff
@@ -100,14 +100,20 @@ async def run(self):
                     if not price_account.aggregate_price_info:
                         raise RuntimeError("Aggregate price info is missing")
 
+                    # When min_publishers is high it means that the price is not production-ready
+                    # yet and it is still being tested. We need no alerting for these prices.
+                    if price_account.min_publishers >= 10:
```
Comment (ali-bahjati) [inline pyth_observer/__init__.py:105]: This is equivalent to "beta" on the website and filters not ready prices.

File: pyth_observer/check/price_feed.py [hunk 2]
```diff
@@ -46,10 +46,11 @@ def error_message(self) -> str:
         ...
 
 
-class PriceFeedAggregateCheck(PriceFeedCheck):
+class PriceFeedOfflineCheck(PriceFeedCheck):
```
Comment (ali-bahjati) [inline pyth_observer/check/price_feed.py:49]: The name was ambiguous. Offline is a better name.

File: pyth_observer/check/price_feed.py [hunk 3]
```diff
@@ -64,28 +65,30 @@ def run(self) -> bool:
         if not is_market_open:
             return True
 
-        # Skip if not trading
-        if self.__state.status != PythPriceStatus.TRADING:
-            return True
-
         distance = abs(
-            self.__state.slot_aggregate_attempted - self.__state.slot_aggregate
+            self.__state.latest_block_slot - self.__state.latest_trading_slot
```
Comment (ali-bahjati) [inline pyth_observer/check/price_feed.py:69]: A price feed can go stale without its status becoming non-trading. This check relies on checking the latest slot with the latest trading slot (which includes non-trading checks before)

File: pyth_observer/check/price_feed.py [hunk 4]
```diff
@@ -64,28 +65,30 @@ def run(self) -> bool:
...
         )
 
         # Pass if distance is less than max slot distance
         if distance < self.__max_slot_distance:
             return True
 
+        # Pass if price has been stale for a long time
+        if distance > self.__abandoned_slot_distance:
```
Comment (ali-bahjati) [inline pyth_observer/check/price_feed.py:77]: Some feeds are abandoned but still in our list.

File: pyth_observer/check/publisher.py [hunk 5]
```diff
@@ -33,11 +36,11 @@ def state(self) -> PublisherState:
     def run(self) -> bool:
         ...
 
-    def error_message(self, publishers: Dict[str, str]) -> str:
+    def error_message(self) -> str:
         ...
 
 
-class PublisherAggregateCheck(PublisherCheck):
+class PublisherWithinAggregateConfidenceCheck(PublisherCheck):
```
Comment (ali-bahjati) [inline pyth_observer/check/publisher.py:43]: Better name again.

File: pyth_observer/check/publisher.py [hunk 6]
```diff
@@ -46,10 +49,23 @@ def state(self) -> PublisherState:
...
+        if self.__state.aggregate_status != PythPriceStatus.TRADING:
+            return True
+
         # Skip if confidence interval is zero
         if self.__state.confidence_interval == 0:
             return True
 
+        # Pass if publisher slot is far from aggregate slot
+        distance = abs(self.__state.slot - self.__state.aggregate_slot)
+        if distance > 25:
```
Comment (ali-bahjati) [inline pyth_observer/check/publisher.py:66]: This value is hard-coded but it is not very important. It is used to filter out the non-trading or behind publishers. We can change it later in the future.

File: pyth_observer/check/publisher.py [hunk 7]
```diff
@@ -141,47 +175,51 @@ def state(self) -> PublisherState:
...
         if self.__state.status != PythPriceStatus.TRADING:
             return True
 
         # Skip if publisher is too far behind
+        slot_diff = abs(self.__state.slot - self.__state.aggregate_slot)
         if slot_diff > self.__max_slot_distance:
             return True
 
-        # Skip if no aggregate
-        if self.__state.price_aggregate == 0:
```
Comment (ali-bahjati) [inline pyth_observer/check/publisher.py:156]: When aggregate status is trading this is impossible.

File: pyth_observer/event.py [hunk 8]
```diff
@@ -36,15 +34,7 @@ def __init__(self, check: Check, context: Context):
...
-        if self.check.__class__.__bases__ == (PublisherCheck,):
-            text = cast(PublisherCheck, self.check).error_message(
-                self.context["publishers"]
-            )
-        elif self.check.__class__.__bases__ == (PriceFeedCheck,):
-            text = cast(PriceFeedCheck, self.check).error_message()
-        else:
-            raise RuntimeError("Invalid check")
-
+        text = self.check.error_message()
```
Comment (ali-bahjati) [inline pyth_observer/event.py:37]: Added publisher_name to publisher state and the code here is therefore simpler. 

Also with the previous logic if a new publisher was added this code would crash as it couldn't look up the name.

**Review by jayantk — APPROVED** [review]
nice!

**Pull request merged**